### PR TITLE
Resolver/fix damage logic

### PR DESCRIPTION
**Pull request by Tyranisaur**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Resolver
 - Fixed: Bug where nested requirements were combined wrongly.
+- Fixed: Bug where damage constraints in chains were not understood correctly.
```
Comment (Miepee) [inline CHANGELOG.md:20]: You should also add a changelog entry that prime doesnt have multiplicative suit reductions anymore.

File: randovania/game_description/game_description.py [hunk 2]
```diff
@@ -213,14 +221,19 @@ def helper() -> Iterator[ResourceInfo]:
...
                     # Ignore those with the `negate` flag. We can't "uncollect" a resource to satisfy these.
                     # Finally, if it's not satisfied then we're interested in collecting it
-                    if not individual.negate and not individual.satisfied(resources, energy, database):
+                    if not individual.negate and not individual.satisfied(resources, current_energy, database):
                         if individual.is_damage:
                             assert isinstance(individual.resource, SimpleResourceInfo)
                             yield from _resources_for_damage(individual.resource, database, resources)
                         else:
                             yield individual.resource
+                    elif individual.is_damage and individual.satisfied(resources, current_energy, database):
```
Comment (henriquegemignani) [inline randovania/game_description/game_description.py:234]: Do you think you could make a test case for this function? It'll be a bit tricker as this function doesn't have tests yet though :(

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

**Review by henriquegemignani — APPROVED** [review]

**Comment by ThanatosGit** [comment]
The usual "new version = changelog conflict" thing.

**Comment by Tyranisaur** [comment]
> The usual "new version = changelog conflict" thing.

I also moved the changelog entries to the correct version :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -112,6 +120,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fixed: Warrior shrine loading behavior
 - Changed: Remove white screen flash effect when crates explode.
 - Changed: Skippable cutscene modes are no longer experimental. Skippable is the new default. Competitive cutscene mode has been updated appropriately.
+
```
Comment (ThanatosGit) [inline CHANGELOG.md:123]: unintentional new line in 6.3.0?

**Pull request merged**